src/Nethermind/Nethermind.Merge.Plugin/MergePlugin.cs

```diff
@@ -106,17 +118,27 @@ public Task InitNetworkProtocol()
                 if (_api.UnclesValidator is null) throw new ArgumentNullException(nameof(_api.UnclesValidator));
                 if (_api.BlockProductionPolicy == null) throw new ArgumentException(nameof(_api.BlockProductionPolicy));
                 if (_api.SealValidator == null) throw new ArgumentException(nameof(_api.SealValidator));
-                
-                _api.HeaderValidator = new MergeHeaderValidator(_poSSwitcher, _api.BlockTree, _api.SpecProvider, _api.SealValidator, _api.LogManager);
+
+                _api.HeaderValidator = new InvalidHeaderInterceptor(
+                    new MergeHeaderValidator(_poSSwitcher, _api.BlockTree, _api.SpecProvider, _api.SealValidator,
+                        _api.LogManager),
+                    _invalidChainTracker,
+                    _api.LogManager);
                 _api.UnclesValidator = new MergeUnclesValidator(_poSSwitcher, _api.UnclesValidator);
-                _api.BlockValidator = new BlockValidator(_api.TxValidator, _api.HeaderValidator, _api.UnclesValidator,
-                    _api.SpecProvider, _api.LogManager);
+                _api.BlockValidator = new InvalidBlockInterceptor(
+                    new BlockValidator(_api.TxValidator, _api.HeaderValidator, _api.UnclesValidator,
+                        _api.SpecProvider, _api.LogManager),
+                    _invalidChainTracker,
+                    _api.LogManager);
                 _api.HealthHintService =
                     new MergeHealthHintService(_api.HealthHintService, _poSSwitcher);
                 _mergeBlockProductionPolicy = new MergeBlockProductionPolicy(_api.BlockProductionPolicy);
                 _api.BlockProductionPolicy = _mergeBlockProductionPolicy;
 
                 _api.FinalizationManager = new MergeFinalizationManager(_blockFinalizationManager, _api.FinalizationManager, _poSSwitcher);
+                
+                // Need to do it here because blockprocessor is not available in init
+                _invalidChainTracker.SetupBlockchainProcessorInterceptor(_api.BlockchainProcessor!);
```

Comment:
Maybe we should add InitBlockchain to plugins that is after InitBlockchain step? @MarekM25 WDYT?